extensions/wrapper/clients/typescript-client/src/oauth2/AccessTokenService.ts

```diff
@@ -0,0 +1,54 @@
+import {ClientCredentialsResponse, fetchClientCredentials} from './FetchUtils';
+import {buildTokenRequestFormData} from './RequestUtils';
+import {ClientCredentials} from './model/ClientCredentials';
+
+export class AccessTokenService {
+    private activeRequest: Promise<ClientCredentialsResponse> | null = null;
+    private refreshTimeout?: NodeJS.Timeout;
+    private tokenRequestFormData: string;
+    private accessToken: string | null = null;
+
+    constructor(private clientCredentials: ClientCredentials) {
+        this.tokenRequestFormData = buildTokenRequestFormData(
+            clientCredentials.clientId,
+            clientCredentials.clientSecret,
+        );
+    }
+
+    async getAccessToken(): Promise<string> {
+        if (!this.accessToken) {
+            return await this.refreshAccessToken();
+        }
+        return this.accessToken;
+    }
+
+    /**
+     * Synchronized refreshing of the access token
+     */
```

Comment:
is this comment still required?

Reply:
Well it add the context that it is synchronized meaning now simultaneous refreshes